.github/workflows/sync-repos.yaml

```diff
@@ -0,0 +1,140 @@
+    # changed since most recent commit to master.
+    paths:
+      - 'stable/*/**'
+      - 'incubator/*/**'
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      stable: ${{ steps.filter.outputs.stable }}
+      incubator: ${{ steps.filter.outputs.incubator }}
+      stable_dirs: ${{ steps.dirs.outputs.stable_dirs }}
+      incubator_dirs: ${{ steps.dirs.outputs.incubator_dirs }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2.5.0
+        id: filter
+        with:
+          list-files: shell
+          filters: |
+            stable:
+              - added|modified: 'stable/*/**'
+            incubator:
+              - added|modified: 'incubator/*/**'
+      # Exports space-separated lists of changed chart dirs
+      #   "xargs -n1" converts to newlines
+      #   "cut -d/ -f 1-2" gets the first two dirs (example: stable/wordpress)
+      #   "sort -u" gets sorted unique dirs
+      #   "xargs" converts back to space-separated string
+      - id: dirs
+        run: |
+          echo "stable files: ${{ steps.filter.outputs.stable_files }}"
+          echo "incubator files: ${{ steps.filter.outputs.stable_files }}"
+          stable_dirs=$(echo ${{ steps.filter.outputs.stable_files }} | xargs -n1 | cut -d/ -f 1-2 | sort -u | xargs)
+          echo "::set-output name=stable_dirs::$stable_dirs"
+          echo "stable dirs: $stable_dirs"
+          incubator_dirs=$(echo ${{ steps.filter.outputs.incubator_files }} | xargs -n1 | cut -d/ -f 1-2 | sort -u | xargs)
+          echo "::set-output name=incubator_dirs::$incubator_dirs"
+          echo "incubator dirs: $incubator_dirs"
+
+  package:
+    needs: changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: package stable
+        if: ${{ needs.changes.outputs.stable == 'true' }}
+        run: |
+          echo "dirs: ${{ needs.changes.outputs.stable_dirs }}"
+          mkdir -p ${{ runner.temp }}/stable/packages
+          for dir in ${{ needs.changes.outputs.stable_dirs }}; do
+              if helm dependency build "$dir"; then
+                  helm package --destination ${{ runner.temp }}/stable/packages "$dir"
+                  echo "✅ built dependencies and packaged $dir"
+              else
+                  echo "❌ unable to build dependencies for $dir"
+                  exit 1
+              fi
+          done
+      - uses: actions/upload-artifact@v2
+        if: ${{ needs.changes.outputs.stable == 'true' }}
+        with:
+          name: stable-packages
+          path: ${{ runner.temp }}/stable/packages/
+      - name: package incubator
+        if: ${{ needs.changes.outputs.incubator == 'true' }}
+        run: |
+          mkdir -p ${{ runner.temp }}/incubator/packages
+          for dir in ${{ needs.changes.outputs.incubator_dirs }}; do
+              if helm dependency build "$dir"; then
+                  helm package --destination ${{ runner.temp }}/incubator/packages "$dir"
```

Comment:
You don need the separate `helm dep build` if you specify `--dependency-update` here.

Reply:
Yep, agreed. I was trying to keep this part of the workflow as close to `repo-sync.sh`. But I can change this.